nautobot/core/models/tree_queries.py

```diff
@@ -15,10 +14,13 @@ def max_tree_depth(self):
         """
         Get the maximum depth of any tree in this queryset.
         """
-        return self.with_tree_fields().extra(order_by=["-__tree.tree_depth"]).first().tree_depth
+        deepest = self.with_tree_fields().extra(order_by=["-__tree.tree_depth"]).first()
+        if deepest is not None:
+            return deepest.tree_depth
+        return 0
```

Comment:
Not sure why this hasn't been an issue previously, but I definitely hit a case while testing this where `max_tree_depth` was being called with no model instances present, hence this fix.